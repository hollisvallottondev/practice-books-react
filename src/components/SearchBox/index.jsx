import React, {useState} from 'react';
import styles from './styles.scss'; 
import PropTypes from 'prop-types';
import CustomInput from '../CustomInput/index.jsx';

const SearchBox = ({handleSearch, handleAdd}) => {
	return (
		<div className={styles.searchBoxWrapper}>
			<div className={styles.searchBoxContainer}>
				<CustomInput 
					handleSearch={handleSearch} 
					icon={'fas fa-search'}
					placeHolder={'Find a book by title or author'}
				/>
			</div>
			<div className={styles.searchBoxContainer}>
				<button className={styles.addButton} onClick={handleAdd}>Add a new book</button>
			</div>
		</div>
	)
}

SearchBox.propTypes = {
	handleSearch: PropTypes.func.isRequired, 
	handleAdd: PropTypes.func.isRequired
}



export default SearchBox;