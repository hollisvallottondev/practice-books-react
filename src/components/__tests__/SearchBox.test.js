import React from 'react';
import SearchBox from '../SearchBox/index.jsx';
import renderer from 'react-test-renderer';


test('SearchBox renders correctly', () => {
	const handleSearch = jest.fn(() => console.log('Handle search'));
	const handleAdd = jest.fn(() => console.log('Handle add'));

	const component = renderer.create(
		<SearchBox 
			handleSearch={handleSearch}
			handleAdd = {handleAdd} />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});