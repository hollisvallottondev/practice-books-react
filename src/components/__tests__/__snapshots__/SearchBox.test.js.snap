// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders correctly 1`] = `
<div>
  <div>
    <div>
      <div>
        <i
          className="fas fa-search"
        />
        <input
          onChange={[Function]}
          placeholder="Find a book by title or author"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div>
    <button
      onClick={[MockFunction]}
    >
      Add a new book
    </button>
  </div>
</div>
`;
